tests/importer/importer.cpp

```diff
@@ -0,0 +1,761 @@
+                                "    <units units_ref=\"units4\" name=\"units4FromLibrary\"/>\n"
+                                "  </import>\n"
+                                "</model>";
+
+    auto funkyModel = parser->parseModel(funkyString);
+    EXPECT_EQ(size_t(0), parser->issueCount());
+
+    // Pass in a base file path, this will be used to resolve URLs if they don't already exist as keys.
+    // This allows the model to be resolved against different locations where local files are specified.
+    importer->resolveImports(funkyModel, resourcePath("importer/"));
+    EXPECT_FALSE(funkyModel->hasUnresolvedImports());
+    EXPECT_EQ(size_t(0), importer->issueCount());
+
+    // Funky hrefs are valid ...
+    auto validator = libcellml::Validator::create();
+    validator->validateModel(funkyModel);
+    EXPECT_EQ(size_t(0), validator->issueCount());
+
+    // ... and can be flattened as expected too.
+    auto flattenedFunkyModel = importer->flattenModel(funkyModel);
+    EXPECT_EQ(size_t(0), importer->issueCount());
+    flattenedFunkyModel->setName("flattenedFunkyModel");
+    validator->validateModel(flattenedFunkyModel);
+    EXPECT_EQ(size_t(0), validator->issueCount());
+}
+
+TEST(Importer, resolveApiModelImports)
+{
+    auto importer = libcellml::Importer::create();
+
+    // Create a source model, and add units and components for use elsewhere.
+    auto source1 = libcellml::Model::create("source1");
+    source1->addComponent(libcellml::Component::create("componentThatINeed"));
+    source1->addUnits(libcellml::Units::create("unitsThatINeed"));
+    auto vanilla = libcellml::Variable::create("vanilla");
+    source1->component(0)->addVariable(vanilla);
+
+    // Create a model which consumes "componentThatINeed" and "unitsThatINeed".
+    auto model = libcellml::Model::create("model");
+    model->addComponent(libcellml::Component::create("componentIWillImport"));
+    model->addUnits(libcellml::Units::create("unitsThatIWillImport"));
+    auto imp1 = libcellml::ImportSource::create();
+    imp1->setUrl("flavourOfTheMonth"); // Set URL to be an arbitrary string.
+    model->component(0)->setImportSource(imp1); // Set the same import source for both units and component
+    model->units(0)->setImportSource(imp1);
+    model->component(0)->setImportReference("componentThatINeed");
+    model->units(0)->setImportReference("unitsThatINeed");
+
+    // Add the source model to the importer library, as "flavour of the month".
+    importer->addModel(source1, "flavourOfTheMonth");
+
+    // Resolve the model against the importer library and check it contains vanilla.
+    importer->resolveImports(model, "");
+    EXPECT_EQ(size_t(0), importer->issueCount());
+    EXPECT_FALSE(model->hasUnresolvedImports());
+    auto flatModel = importer->flattenModel(model);
+    EXPECT_EQ("vanilla", flatModel->component(0)->variable(0)->name());
```

Comment:
Should we also test the units? If not, then why bother with it in the first place?

Reply:
The point of this test is to show how components can be switched around using the importer, so that users can switch out a component for another.  In this case, the test is that the component which is switched changes variables.  The units are the same.

Reply:
Yes, I got that, it's just that you are declaring a units, but not doing anything with it. So, considering the purpose of this test, I would remove all the units-related stuff. To me, it's unnecessary noise in your test.